packages/api/src/share_link/controller.test.ts

```diff
@@ -331,3 +332,217 @@ describe("POST /share-links", () => {
+    await agent
+      .patch("/api/v2/share-links/1000")
+      .send({ maxUses: 100 })
+      .expect(400);
+  });
+
+  test("should return 400 if the request attempts to set permissionsLevel above 'viewer' on an unlisted link", async () => {
+    await agent
+      .patch("/api/v2/share-links/1000")
+      .send({ permissionsLevel: "owner" })
+      .expect(400);
+  });
+
+  test("should return 400 if the request attempts to set a link with maxUses set to be unlisted", async () => {
+    await agent
+      .patch("/api/v2/share-links/1001")
+      .send({ accessRestrictions: "none" })
+      .expect(400);
+  });
+
+  test("should return 400 if the request attempts to set a link where permissionsLevel != 'viewer to be unlisted", async () => {
+    await agent
+      .patch("/api/v2/share-links/1002")
+      .send({ accessRestrictions: "none" })
+      .expect(400);
+  });
+
+  test("should allow maxUses and permissionsLevel != 'viewer' if updating accessRestrictions to something other than 'none'", async () => {
+    await agent
+      .patch("/api/v2/share-links/1000")
+      .send({
+        permissionsLevel: "owner",
+        maxUses: 100,
+        accessRestrictions: "account",
+      })
+      .expect(200);
+  });
+
+  test("should update provided fields", async () => {
+    await agent
+      .patch("/api/v2/share-links/1000")
+      .send({
+        accessRestrictions: "account",
+        permissionsLevel: "editor",
+        maxUses: 500,
+        expirationTimestamp: "2052-01-01T00:00:00.000Z",
+      })
+      .expect(200);
+
+    const shareLink = await retrieveShareLink("5");
+    expect(shareLink?.maxuses).toEqual("500");
+    expect(shareLink?.unrestricted).toEqual(false);
+    expect(shareLink?.autoapprovetoggle).toEqual(1);
+    expect(shareLink?.defaultaccessrole).toEqual("access.role.editor");
+    expect(shareLink?.expiresdt).toEqual(new Date("2052-01-01T00:00:00.000Z"));
+  });
+
+  test("should update nullable fields to null if request sets them to explicit null", async () => {
+    await agent
+      .patch("/api/v2/share-links/1001")
+      .send({
+        maxUses: null,
+        expirationTimestamp: null,
+      })
+      .expect(200);
+
+    const shareLink = await retrieveShareLink("6");
+    expect(shareLink?.maxuses).toBeNull();
+    expect(shareLink?.expiresdt).toBeNull();
+  });
+
+  test("should not update fields if they aren't set in the request", async () => {
+    await agent
+      .patch("/api/v2/share-links/1001")
+      .send({
+        permissionsLevel: "editor",
+      })
+      .expect(200);
+
+    const shareLink = await retrieveShareLink("6");
+    expect(shareLink?.maxuses).not.toBeNull();
+    expect(shareLink?.expiresdt).not.toBeNull();
+    expect(shareLink?.defaultaccessrole).toEqual("access.role.editor");
+    expect(shareLink?.unrestricted).toEqual(false);
+    expect(shareLink?.autoapprovetoggle).toEqual(0);
+  });
+
+  test("should return 400 if the request is empty", async () => {
```

Comment:
Right now this actually sends an empty object (rather than an empty request).  I could see value in testing both, but as named I would expect it to `send()`.